src/frontend/Debug_data_generation.ml

```diff
@@ -2,6 +2,41 @@ open Core_kernel
 open Middle
 open Ast
 
+let rec transpose = function
+  | [] :: _ -> []
+  | rows ->
+      let hd = List.map ~f:List.hd_exn rows in
+      let tl = List.map ~f:List.tl_exn rows in
```

Comment:
Not sure I understand this one. hd :: tl would give me the first row (since rows is a list of lists) and the rest, but I want the head of each individual row and the rest for the transpose. Am I missing something?

Reply:
Oh, right, there's a List.map. Sorry, wasn't paying attention.